File: dispatch/p1_cpp_dispatch/s1_function.cpp
int area_square(int side) {
    return side * side;
}

int area_triangle(int side, int height) {
    return (side * height)/2;
}

int main() {
    auto a1 = area_square(5);
    auto a2 = area_triangle(10, 20);
    return a1 + a2;
}

File: dispatch/p2_rust_dispatch/s1_function.rs
pub fn square_area(side: i32) -> i32 {
    side * side
}

pub fn triangle_area(side: i32, height: i32) -> i32 {
    (side * height)/2
}

pub fn main() -> i32 {
    let v1 = square_area(5);
    let v2 = triangle_area(10, 20);
    v1 + v2
}
